src/primitives.rs

```diff
@@ -0,0 +1,117 @@
+//! Contains comparison primitives used to build up the rest of the library.
+
+/// Returns list of triples describing matching sequences.
+///
+/// The first number is the index in the first string of the beginning of the match.
+/// The second number is the index of the second string of the beginning of the match.
+/// The final number is the length of the match.
+///
+/// The final matching sequence will be a trivial matching sequence of (a.len(),
+/// b.len(), 0) and will be the only match of length 0.
+///
+/// ```
+/// # use fuzzywuzzy::segmentation::{Segmenter, CodePointSegmenter, GraphemeSegmenter};
+/// # use fuzzywuzzy::primitives::get_matching_blocks;
+/// assert_eq!(get_matching_blocks(&CodePointSegmenter.segment("abxcd"), &CodePointSegmenter.segment("abcd")), vec![(0, 0, 2), (3, 2, 2), (5, 4, 0)]);
```

Comment:
Thoughts on doc-tests versus standalone unit tests? I see there is less boilerplate to get started writing the tests, and its likely easier to keep them up-to-date along with the docs since they're co-located, but it seems a maintenance hassle to be writing code in `///` comments (due to the line-length) unless you have an editor that supports this.

Reply:
I agree with everything said. my editor doesn't format these docs which is frustrating.

I'm tending towards doc tests because they are typically also good examples. I can imagine a more complete test suite that we might not want fully exposed, but we don't have that kind of test coverage yet. In general, I'd say doc tests first for the most important properties and examples, then standalone tests for the rest.

Reply:
Makes sense to me! I've always done standalone tests since I've never spent the time to familiarize myself with the specifics of doc-tests (e.g. the reason for prepending `#` before `use` statements, how to create multiple doc-tests for the same section of docs, etc.).

Reply:
the `#` in docs just hides the line so that your imports aren't displayed. It could be omitted to display the import for slightly more verbose displayed code.

`cargo test` just runs all code blocks in docs. So if you want multiple sets of tests, just add additional code blocks!